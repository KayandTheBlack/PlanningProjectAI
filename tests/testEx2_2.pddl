(define (problem autogen)
  (:domain worldsEndAndBeyond)

  (:objects
    Origin cityA cityB cityC cityD cityE cityF cityG cityH cityI cityJ cityK cityL - city
    hotelA_A hotelA_B hotelB_A hotelB_B hotelC_A hotelC_B hotelD_A hotelD_B hotelE_A hotelE_B hotelF_A hotelF_B hotelG_A hotelG_B hotelH_A hotelH_B hotelI_A hotelI_B hotelJ_A hotelJ_B hotelK_A hotelK_B hotelL_A hotelL_B - hotel
  )

  (:init
    (currentLocation Origin)

    (hotelAt hotelA_A cityA)
    (hotelAt hotelA_B cityA)
    (hotelAt hotelB_A cityB)
    (hotelAt hotelB_B cityB)
    (hotelAt hotelC_A cityC)
    (hotelAt hotelC_B cityC)
    (hotelAt hotelD_A cityD)
    (hotelAt hotelD_B cityD)
    (hotelAt hotelE_A cityE)
    (hotelAt hotelE_B cityE)
    (hotelAt hotelF_A cityF)
    (hotelAt hotelF_B cityF)
    (hotelAt hotelG_A cityG)
    (hotelAt hotelG_B cityG)
    (hotelAt hotelH_A cityH)
    (hotelAt hotelH_B cityH)
    (hotelAt hotelI_A cityI)
    (hotelAt hotelI_B cityI)
    (hotelAt hotelJ_A cityJ)
    (hotelAt hotelJ_B cityJ)
    (hotelAt hotelK_A cityK)
    (hotelAt hotelK_B cityK)
    (hotelAt hotelL_A cityL)
    (hotelAt hotelL_B cityL)

    (flight Origin cityA)
    (flight Origin cityB)
    (flight Origin cityC)
    (flight Origin cityD)
    (flight Origin cityE)
    (flight Origin cityF)
    (flight Origin cityG)
    (flight Origin cityH)
    (flight Origin cityI)
    (flight Origin cityJ)
    (flight Origin cityK)
    (flight Origin cityL)
    (flight cityA cityB)
    (flight cityA cityG)
    (flight cityA cityH)
    (flight cityA cityJ)
    (flight cityB cityC)
    (flight cityB cityH)
    (flight cityB cityK)
    (flight cityB cityL)
    (flight cityC cityI)
    (flight cityC cityJ)
    (flight cityC cityK)
    (flight cityC cityL)
    (flight cityD cityA)
    (flight cityD cityB)
    (flight cityD cityK)
    (flight cityD cityL)
    (flight cityE cityA)
    (flight cityE cityH)
    (flight cityE cityI)
    (flight cityE cityK)
    (flight cityF cityA)
    (flight cityF cityB)
    (flight cityF cityH)
    (flight cityF cityK)
    (flight cityG cityA)
    (flight cityG cityB)
    (flight cityG cityF)
    (flight cityG cityH)
    (flight cityH cityE)
    (flight cityH cityG)
    (flight cityH cityI)
    (flight cityH cityK)
    (flight cityI cityC)
    (flight cityI cityE)
    (flight cityI cityF)
    (flight cityI cityJ)
    (flight cityJ cityC)
    (flight cityJ cityE)
    (flight cityJ cityF)
    (flight cityJ cityK)
    (flight cityK cityB)
    (flight cityK cityG)
    (flight cityK cityI)
    (flight cityK cityJ)
    (flight cityL cityB)
    (flight cityL cityF)
    (flight cityL cityG)
    (flight cityL cityJ)

    (= (citiesVisited) 0)
    (= (totalDays) 0)
    (= (minDaysInCity) 1)
    (= (maxDaysInCity) 2)

    (= (stayDuration Origin) 0)
    (= (stayDuration cityA) 0)
    (= (stayDuration cityB) 0)
    (= (stayDuration cityC) 0)
    (= (stayDuration cityD) 0)
    (= (stayDuration cityE) 0)
    (= (stayDuration cityF) 0)
    (= (stayDuration cityG) 0)
    (= (stayDuration cityH) 0)
    (= (stayDuration cityI) 0)
    (= (stayDuration cityJ) 0)
    (= (stayDuration cityK) 0)
    (= (stayDuration cityL) 0)
    (= (cityInterest Origin) 0)
    (= (cityInterest cityA) 1)
    (= (cityInterest cityB) 2)
    (= (cityInterest cityC) 2)
    (= (cityInterest cityD) 2)
    (= (cityInterest cityE) 1)
    (= (cityInterest cityF) 1)
    (= (cityInterest cityG) 2)
    (= (cityInterest cityH) 3)
    (= (cityInterest cityI) 1)
    (= (cityInterest cityJ) 3)
    (= (cityInterest cityK) 3)
    (= (cityInterest cityL) 2)
    (= (totalInterest) 0)

  )

  (:goal
    (and
      (>= (citiesVisited) 6)
      (>= (totalDays) 20)
    )
  )

  (:metric 
    minimize (totalInterest)
  )

)
