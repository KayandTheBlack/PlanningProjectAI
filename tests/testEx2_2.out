
ff: parsing domain file
domain 'WORLDSENDANDBEYOND' defined
 ... done.
ff: parsing problem file
problem 'AUTOGEN' defined
 ... done.


metric established (normalized to minimize): ((1.00*[RF0](TOTALINTEREST)) - () + 0.00)

checking for cyclic := effects --- OK.

ff: search configuration is  best-first on 1*g(s) + 5*h(s) where
    metric is ((1.00*[RF0](TOTALINTEREST)) - () + 0.00)

advancing to distance:   20
                         19
                         18
                         17
                         16
                         15
                         14
                         13
                         12
                         11
                         10
                          9
                          8
                          7
                          6
                          5
                          4
                          3
                          2
                          1
                          0

ff: found legal plan as follows

step    0: FLYANDSTAY ORIGIN CITYH HOTELH_A
        1: FLYANDSTAY CITYH CITYE HOTELE_A
        2: FLYANDSTAY CITYE CITYK HOTELK_A
        3: FLYANDSTAY CITYK CITYJ HOTELJ_A
        4: FLYANDSTAY CITYJ CITYC HOTELC_A
        5: FLYANDSTAY CITYC CITYI HOTELI_A
        6: INCREASESTAY CITYK
        7: FLYANDSTAY CITYI CITYF HOTELF_A
        8: FLYANDSTAY CITYF CITYA HOTELA_A
        9: FLYANDSTAY CITYA CITYG HOTELG_A
       10: INCREASESTAY CITYA
       11: INCREASESTAY CITYC
       12: INCREASESTAY CITYE
       13: INCREASESTAY CITYF
       14: INCREASESTAY CITYG
       15: INCREASESTAY CITYH
       16: INCREASESTAY CITYI
       17: INCREASESTAY CITYJ
       18: FLYANDSTAY CITYG CITYB HOTELB_A
       19: INCREASESTAY CITYB
     

time spent:    0.00 seconds instantiating 133 easy, 0 hard action templates
               0.00 seconds reachability analysis, yielding 38 facts and 132 actions
               0.00 seconds creating final representation with 37 relevant facts, 28 relevant fluents
               0.00 seconds computing LNF
               0.00 seconds building connectivity graph
               0.00 seconds searching, evaluating 229 states, to a max depth of 0
               0.00 seconds total time

